modules/desktop/graphics/ewwbar.nix

```diff
@@ -276,29 +278,43 @@ let
           fi
       }
 
-      set_screen() {
-          brightnessctl set "$1""%" -q
-      }
-
       get() {
-          brightness=$(screen_level)
+          if [ -z "$1" ]; then
+              brightness=$(brightnessctl info | grep -oP '(?<=\().+?(?=%)' | awk '{print $1 + 0.0}')
```

Comment:
Still duplicate code for running the brightnessctl binary.
Maybe have separate internal read() function that only runs the brightnessctl that would "return" the string content for both features.

Reply:
I'd rather leave this for a more general refactor, if needed.
eww-volume and eww-brightness could both be adjusted to provide listening just for the respective values, rather than an entire JSON object of value, icon etc.